hathor/indexes/addresses_rocksdb_index.py

```diff
@@ -0,0 +1,174 @@
+# Copyright 2021 Hathor Labs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import TYPE_CHECKING, Iterable, List, Optional, Set, Tuple
+
+from structlog import get_logger
+
+from hathor.indexes.addresses_index import AddressesIndex
+from hathor.pubsub import HathorEvents
+from hathor.transaction import BaseTransaction
+from hathor.transaction.scripts import parse_address_script
+
+if TYPE_CHECKING:  # pragma: no cover
+    import rocksdb
+
+    from hathor.pubsub import EventArguments, PubSubManager
+    from hathor.transaction import TxOutput
+
+logger = get_logger()
+
+_CF_NAME_ADDRESS_INDEX = b'address-index'
+
+
+class AddressesRocksDBIndex(AddressesIndex):
+    """ Index of inputs/outputs by address.
+
+    This index uses rocksdb and the following key format:
+
+        key = [address][tx.timestamp][tx.hash]
+              |--34b--||--4 bytes---||--32b--|
```

Comment:
Why do we have 34 bytes for address? I thought it was 25 bytes

Reply:
Ahhh we are not saving the bytes, we are saving b58, right?

Reply:
Yes, it's using the "encoded" representation. I think we could use the decoded pubkey part only if there is any noticeble performance advantage.